functions/go/create-setters/createsetters/create_setters_test.go

```
@@ -0,0 +1,304 @@
+package createsetters
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+func TestCreateSettersFilter(t *testing.T) {
+	var tests = []struct {
+		name              string
+		config            string
+		input             string
+		expectedResources string
+		errMsg            string
+	}{
+		{
```

Add test for empty data. It should throw an error.

data: {}